### DISCUSSION TITLE
preventDefault inside passive event listener Angular

### DISCUSSION THREAD

**Discussion by drprocter** [darkroomengineering/lenis#176] [Q&A] [answered]
Hello, I am facing an issue within my angular webapp where my console log keeps on giving me the following error message: preventDefault inside passive event listener invocation
It seems that I cannot override the default browser behaviour and this leads to some slight stuttering which kind of removes the whole purpose of the library. I wanted to ask whether anyone else had faced a similar issue or could give me a clue on how I can prevent this from happening. My first instict was that it has something to do with the way Angular loads things in the DOM and that I needed to make a change in one of the lifecycles but after sevaral failed attempts I am back to square one.

**Comment by axherrm** [comment]
Got the same error with Angular 17.1.2 & Lenis 1.0.33

In Chrome your mentioned error gets spammed when scrolling: 
![image](https://github.com/studio-freight/lenis/assets/74977055/8fc3b829-d730-4cd8-8272-8de219181e4c)

In Firefox just a simple warning on the first scroll:
![image](https://github.com/studio-freight/lenis/assets/74977055/761a8c2b-4f4b-4079-9559-7235370717b1)

In earlier versions of my project those errors/warning did not occur, this is the commit that introduced them: https://github.com/axherrm/WebCVTemplate-core/commit/64ce43022907cde801fca26e1ba0e3ef71fa9e25

Weird thing is I did not make any big changes, mostly only restructured code

**Comment by kalves** [comment, marked as answer]
Hi, 
I was facing the same issue as @axherrm after updating **Angular** from v14 to v16.2.15 & **Lenis v1.1.13**. 
After check, that was an issue related to Angular not Lenis. It's a waste to try to solve this error in your component.

According to the documentation [here](https://v16.angular.io/guide/event-binding#binding-to-passive-events), to prevent this behavior, you need add the following line before import 'zone.js'.

``(window as any).__zone_symbol__UNPATCHED_EVENTS = ['scroll', 'wheel'];``

Solved for me ! 
Hope it helps

**Comment by jsuyog2** [comment]
Hello,

**Angular Version:** 18.2.4  
**Node Version:** 22.3.0

I encountered a similar issue and was able to resolve it by using **zoneless change detection** in Angular. You can try it by following the steps outlined in the official Angular guide on running [Angular without ZoneJS (Zoneless)](https://angular.dev/guide/experimental/zoneless).

This worked perfectly for me! I hope it helps others as well.

**Comment by kalves** [comment]
Hi,

Yes, @jsuyog2, I've just done it. Unfortunately, Angular 18 in zone-less mode is not yet compatible with applications using server-side rendering with Hydration. Currently, zone-less is in an experimental state, so for my case, and for others in the same situation, the workaround I mentioned earlier is still helpful.

Thanks anyway for the advice! ;-)
